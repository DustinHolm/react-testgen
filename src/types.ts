export interface ImportBlock {
    sourceFile: string,
    isInternal: boolean,
    imports: Import[]
}

export interface Import {
    isDefault: boolean,
    name: string,
    element?: ExportableElement
}

export interface Export {
    name: string,
    isDefault: boolean,
    element?: ExportableElement
}

export enum ExportType {
    Class,
    Function,
    Interface,
    Variable
}

export interface ExportableElement {
    type: ExportType,
    returnsJSX: boolean
}

export interface ClassElement extends ExportableElement {
    methods: MethodElement[],
    constructors: Constructor[]
}

export interface MethodElement extends FunctionElement {
    name: string
}

export interface FunctionElement extends ExportableElement {
    returnType: string,
    parameters: Attribute[]
}

export interface InterfaceElement extends ExportableElement {
    attributes: Attribute[]
}

export interface VariableElement extends ExportableElement {
    varType: string,
    varValue: string
}

export interface Attribute {
    name: string,
    type: string
}

export interface Constructor {
    parameters: Attribute[]
}