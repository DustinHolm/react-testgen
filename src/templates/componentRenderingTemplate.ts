import { Attribute } from "../types"
import { getPropName } from "./common"

export const createComponentRenderFunction = (name: string, props: Attribute[]): string => {
    const renderCall = props.length === 0
        ? `\
        <${name} />\
`
        : `\
        <${name}
${props.map(createPropCall).join("\n")}
        />\
`

    return `\
function whenComponentIsRendered() {
    return render(
${renderCall}
    )
}
`
}

const createPropCall = (prop: Attribute): string => {
    return `\
            ${prop.name}={${getPropName(prop.name)}}\
`
}