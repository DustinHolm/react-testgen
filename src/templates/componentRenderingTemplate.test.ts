import { Attribute } from "../types"
import { createComponentRenderFunction } from "./componentRenderingTemplate"

describe("createComponentRenderFunction Tests", () => {
    test("Without props", () => {
        const name = "test"
        const props: Attribute[] = []
        const returnedString = createComponentRenderFunction(name, props)
        expect(returnedString).toMatchSnapshot()
    })

    test("With one prop", () => {
        const name = "test"
        const props: Attribute[] = [{ name: "att_1", type: "type_1" }]
        const returnedString = createComponentRenderFunction(name, props)
        expect(returnedString).toMatchSnapshot()
    })

    test("With multiple props", () => {
        const name = "test"
        const props: Attribute[] = [{ name: "att_1", type: "type_1" }, { name: "att_2", type: "type_2" }]
        const returnedString = createComponentRenderFunction(name, props)
        expect(returnedString).toMatchSnapshot()
    })
})