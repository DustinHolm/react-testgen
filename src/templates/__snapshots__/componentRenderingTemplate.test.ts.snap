// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createComponentRenderFunction Tests With multiple props 1`] = `
"function whenComponentIsRendered() {
    return render(
        <test
            att_1={propAtt_1}
            att_2={propAtt_2}
        />
    )
}
"
`;

exports[`createComponentRenderFunction Tests With one prop 1`] = `
"function whenComponentIsRendered() {
    return render(
        <test
            att_1={propAtt_1}
        />
    )
}
"
`;

exports[`createComponentRenderFunction Tests Without props 1`] = `
"function whenComponentIsRendered() {
    return render(
        <test />
    )
}
"
`;
